src/sif/pedestriancost.cc

```diff
@@ -319,8 +319,23 @@ class PedestrianCost : public DynamicCost {
     // On first pass use the walking speed plus a small factor to account for
     // favoring walkways, on the second pass use the the maximum ferry speed.
     if (pass_ == 0) {
-      return (kSecPerHour * 0.001f) /
-             (kDefaultSpeedFoot * std::min(walkway_factor_, sidewalk_factor_));
+
+      // Determine factor based on all of the factor options
+      float factor = 1.f;
+      if (walkway_factor_ < 1.f) {
+        factor *= walkway_factor_;
+      }
+      if (sidewalk_factor_ < 1.f) {
+        factor *= sidewalk_factor_;
+      }
+      if (alley_factor_ < 1.f) {
+        factor *= alley_factor_;
+      }
+      if (driveway_factor_ < 1.f) {
+        factor *= driveway_factor_;
+      }
+
+      return (speedfactor_ * factor);
```

Comment:
noting that A*s goal estimation heuristic should never overestimate the actual cost to the goal, we landed on this implementation.

Reply:
https://en.wikipedia.org/wiki/Consistent_heuristic <-- that is what i was referring to

Reply:
You probably want to limit how small the factors can be. Making them very small values like 0.1 will force the A* factor to change and performance will drop. Also, it is usually more effective to penalize edges rather than favor them, so to dramatically favor walkways one might try penalizing regular roads (weighting factor > 1.0)

Reply:
The `favor` examples were used to show that it was not working as expected prior to the change. The thinking is that it will be used more for avoiding every separate sidewalk to keep a simpler path and calling out street names for users